### PR TITLE
Issue 557 - agbot policy manager failed initializing when there are 2…

### PR DESCRIPTION
**Pull request by linggao**
… policies with the same name

### CODE REVIEW

File: policy/policy_manager_test.go [hunk 1]
```diff
@@ -52,13 +52,6 @@ func Test_Payloadmanager_init_success2(t *testing.T) {
 	}
 }
 
-func Test_Payloadmanager_dup_policy_name(t *testing.T) {
-
-	if _, err := Initialize("./test/pfduptest/", make(map[string]string), nil, true); err == nil {
```
Comment (dabooz) [inline policy/policy_manager_test.go]: It's ok that you removed this test but you should replace it with a test of the new function.

File: policy/policy_manager.go [hunk 2]
```diff
@@ -175,8 +175,11 @@ func Initialize(policyPath string, arch_synonymns config.ArchSynonyms, workloadR
 	// Setup the callback functions for the policy file watcher. The change notification is the only callback
 	// that should be invoked at this time.
 	changeNotify := func(org string, fileName string, policy *Policy) {
-		numberFiles += 1
-		pm.AddPolicy(org, policy)
+		if err := pm.AddPolicy(org, policy); err != nil {
+			glog.Errorf("Unable to add policy file %v/%v during initialization, error: %v", org, fileName, err)
+		} else {
```
Comment (dabooz) [inline policy/policy_manager.go]: You dont need these changes any more with the new code in PolicyFileChangeWatcher

File: policy/policy_file.go [hunk 3]
```diff
@@ -657,6 +657,24 @@ func (c *Contents) GetPolicyName(org, filename string) string {
 	return ""
 }
 
+// check if another file in the same org has the same policy name as another file.
+// It returns the name of the conflict file.
+func (c *Contents) ConflictsWith(org, filename string, pol *Policy) string {
+	if !c.HasOrg(org) {
+		return ""
+	} else if c.HasFile(org, filename) {
```
Comment (dabooz) [inline policy/policy_file.go]: You dont need this check. The caller has already checked that policy file is not being tracked.

File: policy/policy_manager_test.go (hunk 1 above)
Comment (linggao) [reply policy/policy_manager_test.go]: Will do.

File: policy/policy_manager.go (hunk 2 above)
Comment (linggao) [reply policy/policy_manager.go]: It is do guard other error, just in case.

File: policy/policy_file.go (hunk 3 above)
Comment (linggao) [reply policy/policy_file.go]: There are 2 callers for PolicyFileChangeWatcher(), one is from the policy_manager and the other is from agreementbot. PolicyFileChangeWatcher uses this check to make sure the fileChanged function is not get called and this file is not put into the watcher map.

**Pull request merged**